src/modules/transaction/utils.ts

```diff
@@ -22,60 +25,145 @@ import {
   UpdateTransactionStatusAction
 } from './actions'
 
-// Special flag used to determine transaction hashes to be monitored
-export const TRANSACTION_ACTION_FLAG = '_watch_tx'
-
 export function buildActionRef(transaction: Transaction) {
-  const { actionType, withReceipt, payload } = transaction
-  const buildFunction = withReceipt
-    ? buildTransactionWithReceiptPayload
-    : buildTransactionPayload
+  const { actionType, withReceipt, isCrossChain, payload } = transaction
+
+  let transactionPayload: TransactionPayload
+
+  if (
+    isCrossChain &&
+    transaction.toChainId &&
+    transaction.requestId &&
+    transaction.crossChainProviderType
+  ) {
+    transactionPayload = buildCrossChainTransactionFromPayload(
+      transaction.chainId,
+      transaction.toChainId,
+      transaction.hash,
+      transaction.requestId,
+      payload,
+      transaction.crossChainProviderType
+    )
+  } else {
+    const buildFunction = withReceipt
+      ? buildTransactionWithReceiptPayload
+      : buildTransactionPayload
+    transactionPayload = buildFunction(
+      transaction.chainId,
+      transaction.hash,
+      payload,
+      transaction.chainId
+    )
+  }
+
   return {
     type: actionType,
-    payload: buildFunction(transaction.chainId, transaction.hash, payload)
+    payload: transactionPayload
   }
 }
 
-export function isTransactionAction(action: AnyAction): boolean {
-  return !!getTransactionFromAction(action)
+export function isTransactionAction(
+  action: AnyAction
+): action is ActionWithTransactionPayload {
+  return Boolean(action.payload && action.payload[TRANSACTION_ACTION_FLAG])
+}
+
+export function getTransactionPayloadFromAction(
+  action: ActionWithTransactionPayload
+): TransactionPayload['_watch_tx'] {
+  return action.payload[TRANSACTION_ACTION_FLAG]
+}
+
+export function isTransactionActionCrossChain(action: AnyAction): boolean {
+  return (
```

Comment:
```suggestion
 const { chainId, toChainId } = getTransactionPayloadFromAction(action)
  return (isTransactionAction(action) && !!toChainId && chainId !== toChainId)
```
nit-pick: I think this looks a little bit cleaner and easy to read